cmd/bootstrap/cmd/key.go

```diff
@@ -9,78 +9,120 @@ import (
 	"github.com/spf13/cobra"
 
 	"github.com/onflow/flow-go/cmd/bootstrap/run"
+	"github.com/onflow/flow-go/crypto"
 	model "github.com/onflow/flow-go/model/bootstrap"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/network/p2p"
 )
 
 var (
-	flagRole        string
-	flagAddress     string
+	flagRole    string
+	flagAddress string
+
+	// seed flags
 	flagNetworkSeed []byte
 	flagStakingSeed []byte
+	flagMachineSeed []byte
 )
 
+// keyCmd represents the key command
+var keyCmd = &cobra.Command{
+	Use:   "key",
+	Short: "Generate networking, staking, and machine account keys for a partner node and write them to files. If no key-type ('--networking', '--staking', '--machine') flags are specified, all keys are generated. If any key-type flags are specified, only the specified key types are generated. ",
+	Run:   keyCmdRun,
+}
+
+func init() {
+	rootCmd.AddCommand(keyCmd)
+
+	// required flags
+	keyCmd.Flags().StringVar(&flagRole, "role", "", "node role (can be \"collection\", \"consensus\", \"execution\", \"verification\" or \"access\")")
+	_ = keyCmd.MarkFlagRequired("role")
+	keyCmd.Flags().StringVar(&flagAddress, "address", "", "network address")
+	_ = keyCmd.MarkFlagRequired("address")
+
+	keyCmd.Flags().BytesHexVar(&flagNetworkSeed, "networking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded networking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagStakingSeed, "staking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded staking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagMachineSeed, "machine-seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
 // keyCmdRun generate the node staking key, networking key and node information
 func keyCmdRun(_ *cobra.Command, _ []string) {
+
+	// TODO: We need NodeID of the node to continue checking if the node-info.priv.json exists
+	// TODO: Add paths to log messages for clarity
```

Comment:
The node ID is generated as part of this command, so we don't need to check here. Instead we can check below as a sanity check (see other comment)